tests/unit/EntityId/PrefixingEntityIdParserTest.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace Wikibase\DataModel\Services\Tests\EntityId;
+
+use Wikibase\DataModel\Entity\BasicEntityIdParser;
+use Wikibase\DataModel\Entity\EntityIdParsingException;
+use Wikibase\DataModel\Entity\ItemId;
+use Wikibase\DataModel\Entity\PropertyId;
+use Wikibase\DataModel\Services\EntityId\PrefixingEntityIdParser;
+
+/**
+ * @covers Wikibase\DataModel\Services\EntityId\SuffixEntityIdParser
+ *
+ * @license GPL-2.0+
+ */
+class PrefixingEntityIdParserTest extends \PHPUnit_Framework_TestCase {
+
+	/**
+	 * @dataProvider validInputProvider
+	 */
+	public function testParse( $prefix, $input, $expected ) {
+		$parser = new PrefixingEntityIdParser( $prefix, new BasicEntityIdParser() );
+		$this->assertEquals( $expected, $parser->parse( $input ) );
+	}
+
+	public function validInputProvider() {
+		return array(
+			array( 'wikidata:', 'Q1337', new ItemId( 'wikidata:Q1337' ) ),
+			array( 'foo:', 'P123', new PropertyId( 'foo:P123' ) ),
+			array( 'foo:', 'bar:Q1337', new ItemId( 'foo:bar:Q1337' ) ),
```

Comment:
This is the test case that scares me the most. Let's say I do have two localized Wikibase repositories, "debase" and "enbase". How can code ever assume a prefixed EntityId like "enbase:Q1" is *not* prefixed, and prefix it with "debase:"? What does "debase:enbase:Q1" mean then? What if the inner EntityId parser accepts "enbase:Q1"? What will happen then? Some kind of remote validation? Possibly going in a loop? What's the difference between "debase:enbase:Q1" and "enbase:debase:Q1"?